### PR TITLE
day2

### PR DESCRIPTION
**Pull request by ShayRasooly**

### CODE REVIEW

File: Week1/Day2/Daily Challenge  Lists & Strings/DC_code.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+multiples_list = []
+for i in range(1,length+1):
+    multiples_list.append(number*i)
+print(multiples_list)
+
+#Challenge 2
+user_string = input("Enter your word: ")
+final_string = ''
+for i in range(len(user_string)):
+    if user_string[i] != user_string[i-1] or i ==0:
```
Comment (ShayRasooly) [inline Week1/Day2/Daily Challenge  Lists & Strings/DC_code.py:13]: is there a way to shorten this condition?

File: Week1/Day2/ExerciseXP/code_XP.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
+#Exercise 1 : Set
+my_fav_numbers = set([2,3,5,7,8,10,12,17,21,25])
+my_fav_numbers.update([27,30])
+#Remove the last number --- We can do it only if we know the value of rhe "last" item, because set does not have indexes
+# For ex:my_fav_numbers.remove(30)
+
+friend_fav_numbers = set([3, 11, 18])
+# Concatenate my_fav_numbers and friend_fav_numbers to a new variable called our_fav_numbers
+our_fav_numbers = my_fav_numbers|friend_fav_numbers
```
Comment (ShayRasooly) [inline Week1/Day2/ExerciseXP/code_XP.py:9]: amazing! youre the only one that used this operator

File: Week1/Day2/ExerciseXP/code_XP.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+#or using list comprehensions
+floats_list = [i/2 for i in range(3,11)]
+
+#Exercise 5: For Loop
+for i in range(1,21):
+    print(i)
+print()
+for i in range(1,21):
+    if i % 2 == 0:
```
Comment (ShayRasooly) [inline Week1/Day2/ExerciseXP/code_XP.py:47]: what is the difference between a number and its index?

File: Week1/Day2/ExerciseXP/code_XP.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        print(i)
+
+#Exercise 6 : While Loop
+my_name = 'mikhail'
+while True:
+    user_name = input('Enter your name: ').lower()
+    if my_name==user_name:
```
Comment (ShayRasooly) [inline Week1/Day2/ExerciseXP/code_XP.py:54]: any way to avoid this break?

File: Week1/Day2/ExerciseXP/code_XP.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+pizza_toppings=[]
+while True:
+    user_input = input('Please enter pizza topping here (enter "quit" when you finish): ').lower()
+    if user_input != 'quit':
+        pizza_toppings.append(user_input)
+        print(f'I added {user_input} topping to their pizza')
+    else:
+        break
+total_price = 10+len(pizza_toppings)*2.5
+print(f'Your toppings: ',*pizza_toppings,f'Total price is {total_price}',sep='\n')
```
Comment (ShayRasooly) [inline Week1/Day2/ExerciseXP/code_XP.py:75]: nice usage of * operator

File: Week1/Day2/ExerciseXP/code_XP.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+#teenager
+teenager_names = ['Andrew', 'Joseph', 'Daniel', 'Roni','Marcel','Moshe','Kanan','Mikhail']
+tenn_restr = []
+for item in teenager_names:
+    age = int(input(f'Enter your age,{item}: '))
+    if age <= 21:
+        tenn_restr.append(item)
+for item in tenn_restr:
+    teenager_names.remove(item)
```
Comment (ShayRasooly) [inline Week1/Day2/ExerciseXP/code_XP.py:100]: any way to remove without loop?

File: Week1/Day2/ExerciseXP/code_XP.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+#Exercise 10 : Sandwich Orders
+sandwich_orders = ["Tuna sandwich", "Pastrami sandwich", "Avocado sandwich", "Pastrami sandwich", "Egg sandwich", "Chicken sandwich", "Pastrami sandwich"]
+while "Pastrami sandwich" in sandwich_orders:
+    sandwich_orders.remove("Pastrami sandwich")
+print(sandwich_orders)
+
+finished_sandwiches = []
+while sandwich_orders != []:
+    finished_sandwiches.append(sandwich_orders[0])
+    sandwich_orders.remove(sandwich_orders[0])
```
Comment (ShayRasooly) [inline Week1/Day2/ExerciseXP/code_XP.py:112]: more efficient way to remove from list based on index?

File: Week1/Day2/Exercises XP Gold/Gold_code.py [hunk 8]
```diff
@@ -0,0 +1,155 @@
+#Exercise 1: Concatenate Lists
+list1 = [1,2,3,4,5,6,7,8]
+list2 = [1,2,3,4,5,6,7,8]
+list1.extend(list2)
```
Comment (ShayRasooly) [inline Week1/Day2/Exercises XP Gold/Gold_code.py:4]: is there a second way?

File: Week1/Day2/Daily Challenge  Lists & Strings/DC_code.py (hunk 1 above)
Comment (TiurinMikhail) [reply Week1/Day2/Daily Challenge  Lists & Strings/DC_code.py:13]: like this?:
user_string = input("Enter your word: ")
final_string = user_string[0]
for i in range(1,len(user_string)):
    if user_string[i] != user_string[i-1]:
        final_string += user_string[i]
print(final_string)

File: Week1/Day2/ExerciseXP/code_XP.py (hunk 2 above)
Comment (TiurinMikhail) [reply Week1/Day2/ExerciseXP/code_XP.py:9]: Thank you!

File: Week1/Day2/ExerciseXP/code_XP.py (hunk 3 above)
Comment (TiurinMikhail) [reply Week1/Day2/ExerciseXP/code_XP.py:47]: So, the difference between the number and its index is the numerical value of the index itself, which represents the position of the number within the sequence. But here we defined the index range that starts from 1 to 20, so in this case theay are equal. But if we made a list before loop for we will have another situation:
list = [i for i in range(1,21)]
for i in range(len(list)):
         if i % 2 == 0:
               print(list[i])

File: Week1/Day2/ExerciseXP/code_XP.py (hunk 4 above)
Comment (TiurinMikhail) [reply Week1/Day2/ExerciseXP/code_XP.py:54]: my_name = 'mikhail'
user_name = ''
while my_name!=user_name:
    user_name = input('Enter your name: ').lower()

File: Week1/Day2/ExerciseXP/code_XP.py (hunk 6 above)
Comment (TiurinMikhail) [reply Week1/Day2/ExerciseXP/code_XP.py:100]: teenager_names = ['Andrew', 'Joseph', 'Daniel', 'Roni','Marcel','Moshe','Kanan','Mikhail']
tenn_not_restr = []
for item in teenager_names:
    age = int(input(f'Enter your age,{item}: '))
    if age > 21:
        tenn_not_restr.append(item)
teenager_names = tenn_not_restr
print(teenager_names)

File: Week1/Day2/ExerciseXP/code_XP.py (hunk 7 above)
Comment (TiurinMikhail) [reply Week1/Day2/ExerciseXP/code_XP.py:112]: while sandwich_orders:
    sandwich = sandwich_orders.pop()
    finished_sandwiches.append(sandwich)
#or
while sandwich_orders:
    sandwich = sandwich_orders.pop(0)  # Remove the first element
    finished_sandwiches.append(sandwich)

File: Week1/Day2/Exercises XP Gold/Gold_code.py (hunk 8 above)
Comment (TiurinMikhail) [reply Week1/Day2/Exercises XP Gold/Gold_code.py:4]: concatenated_list = [elem for sublist in [list1, list2] for elem in sublist]
print(concatenated_list)

**Pull request merged**